### PR TITLE
JWPlayer Implementation

### PR DESCRIPTION
**Pull request by spra85**

### CODE REVIEW

File: elements/bulbs-video/plugins/google-analytics.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  return `${gaPrefix}.send`;
+};
+
+export default {
+  init (player, gaPrefix) {
+    let context = {
+      player: player,
+      gaPrefix: gaPrefix,
+    };
+    player.on('play', this.onPlay.bind(context));
```
Comment (collin) [inline elements/bulbs-video/plugins/google-analytics.js]: I'm not sure it makes sense to bind context as `this`.
Could drop the `context` structure entirely and bind `player` and `gaPrefix` as arguments.
```js
this.onPlay.bind(this, player, gaPrefix);
```

So onPlay changes to:

```js
onPlay (player, gaPrefix, event) {
  if (!player.playedOnce) {
/* and so on */
```

**Comment by spra85** [comment]
@collin @kand @daytonn @MelizzaP this is ready to roll.  Test links below.  It relies on changes in this PR on Onion Studios endpoint (https://github.com/theonion/videohub/pull/281).  Planning to release as a major breaking change.  Test links: 

Flash VPAID Preroll example (not skippable due to their tags):
http://jw-player.test.onionstudios.com/videos/ways-to-ensure-a-peaceful-sleep-for-your-baby-4116?xgid=e152c307-829a-4e3c-a158-e9a8e0869745

HTML5 Vast Preroll example (skippable):
http://jw-player.test.onionstudios.com/videos/ways-to-ensure-a-peaceful-sleep-for-your-baby-4116?xgid=48b5e950-9e96-452b-9ec8-564a9d91823a

Onion video page example:
http://jw-player.test.theonion.com/video/onion-reviews-finding-dory-53103

Onion Homepage example (scroll to bottom):
http://jw-player.test.theonion.com/

ClickHole video page example:
http://jw-player.test.clickhole.com/video/its-beautiful-and-its-all-her-4529?xgid=e152c307-829a-4e3c-a158-e9a8e0869745

File: elements/bulbs-video/components/revealed.js [hunk 2]
```diff
@@ -93,62 +96,133 @@ export default class Revealed extends React.Component {
     }
 
     this.props.video.tags.forEach(function (tag) {
+      // Temporary until videojs_options completely removed from Onion Studios
```
Comment (kand) [inline elements/bulbs-video/components/revealed.js]: Should this be prefixed with `TODO`?

File: elements/bulbs-video/components/revealed.js [hunk 3]
```diff
@@ -93,62 +96,133 @@ export default class Revealed extends React.Component {
...
+  }
+
+  makeVideoPlayer (element, videoMeta) {
+    element.id = videoMeta.gaPrefix;
+    let player = global.jwplayer(element);
+
+    player.videoMeta = videoMeta;
+
+    player.setup({
+      'key': 'qh5iU62Pyc0P3L4gpOdmw+k4sTpmhl2AURmXpA==',
```
Comment (kand) [inline elements/bulbs-video/components/revealed.js]: Is the linter saying we need quotes around key names?

File: elements/bulbs-video/plugins/google-analytics.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    player.on('adBlock', this.onAdBlock.bind(this));
+    player.on('complete', this.onComplete.bind(this));
+    player.on('time', this.onTime.bind(this));
+    player.on('fullscreen', this.onFullScreen.bind(this));
+    // Leave out for now, this is noisy
+    // player.on('resize', this.onResize.bind(this));
+    player.on('firstFrame', this.onFirstFrame.bind(this));
+    player.on('adSkipped', this.onAdSkipped.bind(this));
+    player.on('adError', this.onAdError.bind(this));
+    // .on('error')
```
Comment (kand) [inline elements/bulbs-video/plugins/google-analytics.js]: Should these be removed or prefixed with `TODO`?

**Comment by kand** [comment]
👍

**Comment by daytonn** [comment]
seems legit 👍

**Comment by collin** [comment]
@spra85 merging #95 into this branch will clean up most of the linter errors

**Comment by collin** [comment]
Otherwise 👍

File: elements/bulbs-video/components/revealed.js (hunk 3 above)
Comment (spra85) [inline elements/bulbs-video/components/revealed.js]: Nope, not sure why I did that for each one. will remove

**Comment by Joyeric2** [comment]
Dearest one
My name is Joy Eric how are you doing today and how about your health,hope you are fine and good.Any way after going through your profile in this site i became interested in you.I will like to have you as my good friend whom i will like to share my life experience with.Your sex, age, race nor distance does not matter to me rather what matters most in a relationship is the maturity,truth and honest that exist between friends.So i will like you to contact me back through my private mail address(joyeric688@gma!l.com )So that i can tell you more about my self and send you more photos of me.Yours Sincerely,

**Pull request merged**